cmd/resolve.go

```diff
@@ -0,0 +1,672 @@
+/*
+ * Copyright 2018-2020 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/nats-io/jwt"
+	"github.com/nats-io/nsc/cmd/store"
+	"github.com/spf13/cobra"
+)
+
+func createProfileCmd() *cobra.Command {
+	var params ProfileCmdParams
+	var cmd = &cobra.Command{
+		Use:   "profile",
+		Short: "Generate a profile from nsc 'URL' that can be used by tooling",
+		Example: `profile nsc://operator
+resolve nsc://operator/account
+resolve nsc://operator/account/user
+resolve nsc://operator/account/user?operatorSeed&accountSeed&userSeed
+resolve nsc://operator/account/user?operatorKey&accountKey&userKey
+resolve nsc://operator?key&seed
+resolve nsc://operator/account?key&seed
+resolve nsc://operator/account/user?key&seed
+resolve nsc://operator/account/user?store=/a/.nsc/nats&keystore=/foo/.nkeys
+
+Output of the program looks like:
+{
+  "user_creds": "<filepath>",
+  "operator" : {
+     "service": "hostport"
+   }
+}
+The user_creds is printed if an user is specified
+Other options (as query string arguments):
+keystore=<dir> that specifies the location of the keystore
+
+store=<dir> that specifies a directory that contains the named operator
+
+[user|account|operator]Key - includes the public key for user, account, 
+operator, If no prefix (user/account/operator is provided, it targets 
+the last object in the configuration path)
+
+[user|account|operator]Seed=<optional public key> - include the seed for 
```

Comment:
list an example with the public key? I also can't find a unit test for this

Reply:
done